docs/reference/indices/diskusage.asciidoc

```diff
@@ -0,0 +1,182 @@
+[[indices-disk-usage]]
+=== Analyze index disk usage API
+++++
+<titleabbrev>Analyze index disk usage</titleabbrev>
+++++
+
+experimental[]
+
+Analyzes the disk usage of each field of an index or data stream.
+This API might not support indices created in previous {es} versions.
+The result of a small index can be inaccurate as some parts of an index
+might not be analyzed by the API.
+
+[source,console]
+--------------------------------------------------
+POST /my-index-000001/_disk_usage?run_expensive_tasks=true
+--------------------------------------------------
+// TEST[setup:messages]
+
+[[analyze-index-disk-usage-api-request]]
+==== {api-request-title}
+
+`POST /<target>/_disk_usage`
+
+[[analyze-index-disk-usage-api-request-prereqs]]
+==== {api-prereq-title}
+
+* If the {es} {security-features} are enabled, you must have the `manage`
+<<privileges-list-indices,index privilege>> for the target index, data stream,
+or alias.
+
+[[analyze-index-disk-usage-api-path-params]]
+==== {api-path-parms-title}
+
+`<target>`::
+(Required, string) Comma-separated list of data streams, indices, and aliases
+used to limit the request. It's recommended to execute this API with a single
+index (or the latest backing index of a data streams) as the API consumes
+resources significantly.
+
+[[analyze-index-disk-usage-api-query-params]]
+==== {api-query-parms-title}
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=allow-no-indices]
++
+Defaults to `true`.
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=expand-wildcards]
++
+Defaults to `open`.
+
+`flush`::
+(Optional, Boolean) If `true`, the API performs a flush before analysis. If
+`false`, the response may not include uncommitted data. Defaults to `true`.
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=index-ignore-unavailable]
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=timeoutparms]
+
+`run_expensive_tasks`::
+(Required, Boolean) Analyzing field disk usage is resource-intensive. To use the
```

Comment:
Just to clarify, does it mean that `_disk_usage` will not run at all if `run_expensive_tasks` is false? If yes, than what's the need of this parameter?  Is `run_expensive_tasks` parameter is some kind extra defence mechanism?

Reply:
>  Is run_expensive_tasks parameter is some kind extra defence mechanism

Yes, that's the purpose. It requires an acknowledge from users.